### PR TITLE
Jupyter Branding Update & User Interface Pass

### PR DESCRIPTION
**Pull request by cameronoelsen**
With the release of the brand guidelines this past week, I have updated the IRkernel site to conform to the guidelines. We unfortunately can't keep the Jupyter logo with the R in the middle of chompy. I had added the "powered by Jupyter" logo on the landing page.

In addition, I saw that some of the material elements on the website werent being used correctly (such as the FAB button process at the top of the page) so I restructured the navigation to be more user friendly. To substitute for the process that was described at the top, I have added material design raised buttons at either the top or bottom of each page to lead the user through the process still.

Here are some screenshots of before and after:

#### Before
screen shot 2016-12-08 at 9 16 48 am

#### After
screen shot 2016-12-08 at 9 15 30 am

#### Before
screen shot 2016-12-08 at 9 16 53 am

#### After
screen shot 2016-12-08 at 9 15 37 am

#### Before
screen shot 2016-12-08 at 9 17 08 am

#### After
screen shot 2016-12-08 at 9 15 41 am

#### Before
screen shot 2016-12-08 at 9 17 13 am

#### After
screen shot 2016-12-08 at 9 15 45 am

#### Before
screen shot 2016-12-08 at 9 17 16 am

#### After
screen shot 2016-12-08 at 9 15 48 am

### CODE REVIEW

File: images/PoweredbyJupyter-darkborder.svg
Comment (flying-sheep) [inline images/PoweredbyJupyter-darkborder.svg:4]: i don’t like the dark border. do you have a material-design compliant logo?

File: images/PoweredbyJupyter-darkborder.svg
Comment (flying-sheep) [inline images/PoweredbyJupyter-darkborder.svg:112]: The brand guidelines say:

> If you have a [thing] that uses [Jupyter software] or relies on its architecture, we recommend […]

This makes clear that it applies to software that *uses* Jupyter components; e.g. PyCharm has a notebook frontend powered by Jupyter, and projects like [atom-notebook](https://github.com/jupyter/atom-notebook) and [hydrogen](https://atom.io/packages/hydrogen) use various parts of its architecture to run code and get output.

However, this doesn’t apply to us: Language kernels are the fundamental part of Jupyter’s architecture. They don’t use anything, they just provide a Jupyter protocol compliant interface to a programming language. (Nitpickers might say that’s a part of its architecture, but keep reading.)

The true relationship is that Jupyter software (can) use the IRkernel; All parts of Jupyter are in the end “powered by a kernel”.

I understand the need to distinguish projects that aren’t directly part of the jupyter project (Add-ons, so to speak), but “powered by” doesn’t fit for language kernel projects.

File: _layouts/raw.html
Comment (flying-sheep) [inline _layouts/raw.html:31]: 1 tab per indentation level pls

File: index.html
Comment (flying-sheep) [inline index.html]: doesn’t look good just left-aligned below the header.

also i like the idea of having our own logo. where in the brand guidelines does it say we can’t have one using elements from the jupyter logo?

File: _layouts/raw.html
Comment (flying-sheep) [inline _layouts/raw.html:59]: I think those are a bit hidden just being in the menu, don’t you think?

File: _layouts/raw.html
Comment (flying-sheep) [inline _layouts/raw.html:44]: where is a link to the docs?

**Review by flying-sheep — CHANGES_REQUESTED** [review]
The term “powered by” doesn’t fit, and I like the idea of a own logo.

The new layout is clean, but has some omissions.

File: index.html
Comment (takluyver) [reply index.html]: I don't know if we've stipulated anything about using the elements (Cameron?), but we have certainly said that we don't want people to put other elements/text inside the Jupyter planet logo. We looked at the IRkernel logo in the discussion about that and agreed to ask to change it.

**Comment by takluyver** [comment]
@cameronoelsen do you have a screenshot of the site with the changes, for those of us feeling lazy? ;-)

Also, any ideas about a logo for IRkernel now that putting elements in chompy's mouth is against the spec?

File: index.html
Comment (flying-sheep) [reply index.html]: sure, let’s change it! i’m not too attached to it. but i’d still like to have a logo that uses Jupyter’s visual language.

for the new logo: if not chompy then maybe another recognizable part? have the R logo orbit the jupyter logo as biggest satellite? in any case, i like the fact that R’s logo fits Jupyter’s thematically, as has a halo like saturn… DAMN, we could have named the kernel “satuRn” 😆

**Comment by flying-sheep** [comment]
the problem here is that this decision should actually be mentioned in the guidelines (the PDF). i don’t see it anyway in there (you mentioned a spec. does a separate spec from that PDF exist?)

i think you need to specify what goes and what doesn’t, and i’d really like to have the designers’ help with the new IRkernel logo.

**Comment by cameronoelsen** [comment]
@takluyver I have updated the PR with screenshots of before and after 👍  Also, in regards to the brand guidelines, @flying-sheep it is stated on page 7 to not change the jupyter logo in any way:
screen shot 2016-12-08 at 9 22 32 am

As for the docs page being omitted, I'll fix that right now! Completely blew past me when I was working on the site, just because I didn't realize that the "docs" text in the navbar was a link to a page 😆  

In regards to getting rid of the current IRkernel logo, I or one of the designers here at Jupyter Cal Poly would love to take a design pass at a new logo for IRkernel as well, I can start on that today and hopefully have it soon.

**Comment by takluyver** [comment]
Thanks Cameron. Should we clarify that we don't want people placing *anything* inside the planet - not just words? We might also want to say something about people using pieces of the logo (I assume we want people to use it in as an indivisible unit?).

**Comment by cameronoelsen** [comment]
@takluyver Yeah I will also make those refinements to the brand guidelines in the next day (have to get the merge in and get the syntax right) to make sure that it is clear that the Jupyter logo shouldn't be changed at all.

File: _layouts/raw.html
Comment (cameronoelsen) [reply _layouts/raw.html:44]: Just fixed this, here is an updated version of what the site looks like with the documentation link included:

screen shot 2016-12-08 at 9 38 01 am

screen shot 2016-12-08 at 9 38 08 am

File: images/PoweredbyJupyter-darkborder.svg
Comment (cameronoelsen) [reply images/PoweredbyJupyter-darkborder.svg:4]: Here is another two variations we could use for it:

### Dark background with white jupyter text
screen shot 2016-12-08 at 9 46 21 am

### No background or border
screen shot 2016-12-08 at 9 47 37 am

My design preference is definitely the powered by logo with the dark border but definitely want to know what you like as well!

Comment (flying-sheep) [reply images/PoweredbyJupyter-darkborder.svg:4]: we could actually put the dark one into the footer! (after removing the backgound to make sure it’s the exact same color)

and a new, compliant logo into the header

File: _layouts/raw.html
Comment (flying-sheep) [reply _layouts/raw.html:44]: great, thanks!

File: images/PoweredbyJupyter-darkborder.svg
Comment (cameronoelsen) [reply images/PoweredbyJupyter-darkborder.svg:4]: screen shot 2016-12-08 at 10 01 37 am

File: _layouts/raw.html
Comment (cameronoelsen) [reply _layouts/raw.html:59]: I think that the slideout menu has become second nature to users now (so they will ultimately find it on the page, I can user test it if you'd like) and I think it's secondary content to everything else on the site. If not, I can try to find another place but I thought this was a nice way to organize the repos so it didn't clutter the content on the page

Comment (takluyver) [reply _layouts/raw.html:59]: I'm not particularly fond of hamburger menus for desktop UIs, though I accept they're a useful convention on smaller phone screens. Maybe the Github links could go at the bottom of the homepage? Design's not my thing, though, so if you think the hamburger menu is the best option even on large screens, I'll bow to your expertise. ;-)

Comment (cameronoelsen) [reply _layouts/raw.html:59]: @takluyver Im not a huge fan as well of the desktop hamburger menu (the breakpoint for it seems to be around 1000px) but I think that this breakpoint is defined in the material design light theme. I also noticed that with the nav items, around the same time when the breakpoint happens, overflow onto the title text on the left. I can try and make it so that the desktop navbar stays at the top longer right now

**Comment by flying-sheep** [comment]
> shouldn't be changed at all

“at all” is probably also not what you’re going for. i’d like to see it composed in some way, or the small satellites changed/animated or something.

**Comment by cameronoelsen** [comment]
@flying-sheep What about a logo like this?
initial

**Comment by flying-sheep** [comment]
maybe keep the halo in orange, but i also like it as it is!

**Comment by cameronoelsen** [comment]
Here is the site with the logo in it and the powered by jupyter logo at the bottom right of the page!
screen shot 2016-12-08 at 10 56 27 am

**Comment by flying-sheep** [comment]
cool! would you mind if i changed “powered by” to “powering” after merging this? 😉 

i’m still convinced that “powered by” doesn’t fit language kernels; neither taken literally nor when going by the description in the style guide.

**Comment by cameronoelsen** [comment]
@flying-sheep I can't do that unless we get approval from the entire design team and a pass by @ellisonbg unfortunately :/ Requires a lot more discussion from the team

File: _layouts/raw.html
Comment (flying-sheep) [reply _layouts/raw.html:59]: as you wish. optimally, a sufficiently wide screen would either have a permanently open sidebar or all the links in the top bar.

a better way to navigate the docs would also be cool, but don’t bother with it for this PR. maybe you can come up with an idea and i’ll try my hand after merging this

**Comment by flying-sheep** [comment]
@cameronoelsen hmm. is there anything else we can do for now so this can get merged? would the plain jupyter logo (with or withoud wordmark) or no jupyter logo be OK until then?

**Comment by cameronoelsen** [comment]
@flying-sheep Just took out the "Powered by Jupyter" logo. Here is what the site now looks like:

screen shot 2017-01-17 at 9 33 40 am

**Comment by ellisonbg** [comment]
@cameronoelsen as our brand guidelines (and trademark policy) are written now, we don't allow modifying the jupyter logo. That would both include putting different text inside the jupyter logo as well as taking other elements of the logo (such as the moons) and using them as elements on other logos. Because of that I am think the IRKernel logo should be visually distinct and not re-use elements of the jupyter logo (moons or otherwise) in its design.

**Comment by flying-sheep** [comment]
OK! then let’s use another logo.

could we ask the the jupyter design team to talk about this in their next meeting?

i’d like it to be recognizable as belonging to jupyter.

an idea to explore here is that the background of the R logo fits the sidereus nuncius figure for jupiter that inspired the jupyter logo’s shells:

**Comment by flying-sheep** [comment]
in other news, i would still like to have a modified “powered by ȷupyter” badge in the corner (with text like “a kernel for ȷupyter” or “powering ȷupyter”).

can i just create that one? if this is not possible via the design guidelines i propose to change them accordingly. as said: “powered by” isn’t correct, but this project is still complementary for jupyter, which means *a* badge would be fitting.

**Comment by flying-sheep** [comment]
thank you @cameronoelsen, the new navigation is much better! and i also liked the logo even though we may not keep it.

merged. i also sqashed the commits and fixed the style/indentation, and removed some spurious elements

**Comment by flying-sheep** [comment]
this is a temporary logo. @cameronoelsen @ellisonbg @takluyver: could you please tell me whom to ask to design a logo?

i think as a jupyter project, it should’t get my bad programmer art.

**Comment by takluyver** [comment]
@cameronoelsen is one of our designers. He may be able to work on a logo himself, or ask one of the other people on the design team to look into it.

**Pull request closed without merging**
